### ISSUE TITLE
Some constraints are ignored in Integer Programming Problem

### ISSUE THREAD

**Issue by zufra** [coin-or/pulp#377]
Details for the issue
--------------------

#### What did you do?
Created a Linear Programming Problem with several constraints and where all the variables are boolean.

#### What did you expect to see?
A solution where all the constraints are satisfied.

#### What did you see instead?
Some constraints are not satisfied even though an optimal solution is supposedly found.

Useful extra information
-------------------------

I explained my problem in more detail [here](https://stackoverflow.com/questions/65471673/pulp-integer-programming-constraint-ignored).

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [x] Windows: ( _version: 10 64bits )
- [ ] Linux: ( _distro:_ ___ )
- [ ] Mac OS: ( _version:_ ___ )
- [ ] Other: ___

#### I'm using python version:

- [ ] 2.7
- [ ] 3.4
- [ ] 3.5
- [ ] 3.6
- [x] Other: 3.8

#### I installed PuLP via:

- [x] pypi (pip install pulp)
- [ ] github
- [ ] Other: ___

#### Did you also

- [ ] Tried out the latest github version: https://github.com/coin-or/pulp
- [x] Searched for an existing similar issue: https://github.com/coin-or/pulp/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by pchtsp** [comment]
Please provide a mwe (https://stackoverflow.com/help/minimal-reproducible-example). 

Alternatively, you can put your exported model:

https://coin-or.github.io/pulp/guides/how_to_export_models.html

Without that, it will be impossible to help.

**Comment by zufra** [comment]
Here is the json file obtained when problem.toJson() is called.

[debug.txt](https://github.com/coin-or/pulp/files/5749427/debug.txt)

**Comment by pchtsp** [comment]
Could you please resend the file **after** having solved it? That way I can check the status of the problem and the values of the variables. I tried to solve it with CPLEX but it apparently does not solve in less than 120 seconds...

**Comment by zufra** [comment]
Here is the json file when problem.toJson() is called **after** the problem is solved.

[debugSolution.txt](https://github.com/coin-or/pulp/files/5752076/debugSolution.txt)

The problem took 628 seconds to be solved partly because there are several variables and restrictions and partly because the code is poorly optimized as I am focusing on creating the correct linear equations problem first.

**Comment by pchtsp** [comment]
In the SO description I think you're mixing x_2_7 and x_27, not sure if this has anything to do with anything.

The constraint is working correctly. Find below the analysis:

```python 
import pulp as pl
import pytups as pt

path = 'debugSolution.txt'

# import model
_vars, prob = pl.LpProblem.from_json(path)

# get all variables with non-zero value
vars_value = pt.SuperDict(_vars).vfilter(pl.value)

# export the lp
prob.writeLP('/home/pchtsp/Downloads/debugSolution.lp')

# the constraint you show in the SO problem is:
# _C3833: - x_2_3 + x_2_7 + x_3_7 <= 1

'x_2_7' in vars_value
# True, so x_2_7 has value 1

'x_3_7' in vars_value
# False, so x_3_7 has value 0

'x_2_3' in vars_value
# False, so x_2_3 has value 0

```

So `-0  + 1 + 0 <= 1` => the constraint is respected.

**Comment by zufra** [comment]
You are absolutely right, I printed each variable with its name and value and the constraint was respected. The problem seems to be that I thought `problem.variables()` returned the variables in the order they were defined and thus I could get a list with all the variables values in ascending order (x_0_0, x_0_1, etc) the following way:

`results = problem.variables()
    results_values = [result.varValue for result in results]`

I used this list to create the excel file I showed in my SO question.

What confused me was that the code worked fine for a smaller problem, but it was probably just a coincidence.

I am very sorry to have wasted your time when the error was in my code, and I am very thankful for your help!
